internal/ethapi/api.go

```diff
@@ -1522,6 +1548,54 @@ func (s *PublicTransactionPoolAPI) SendRawTransaction(ctx context.Context, encod
 	return SubmitTransaction(ctx, s.b, tx)
 }
 
+// SendBlockBatches will:
+// * Verify the batches are signed by the BlockBatchesSender
+// * Update the Geth timestamp to the provided timestamp
+// * handle the RollupTransaction Batches contained in the provided Block atomically
+func (s *PublicTransactionPoolAPI) SendBlockBatches(ctx context.Context, messageAndSig []hexutil.Bytes) []error {
+	if len(messageAndSig) != 2 {
+		return []error{fmt.Errorf("Incorrect number of arguments. Expected 2, got %d", len(messageAndSig))}
+	}
+	if !crypto.VerifyMessageSignature(s.b.ChainConfig().BlockBatchesSender.Bytes(), messageAndSig[0], messageAndSig[1]) {
+		return []error{fmt.Errorf("Signature does not match Block Batch Sender address %x", s.b.ChainConfig().BlockBatchesSender)}
+	}
+	var blockBatches BlockBatches
+	if err := json.Unmarshal(messageAndSig[0], &blockBatches); err != nil {
+		return []error{fmt.Errorf("Incorrect format for BlockBatches type. Received: %s", messageAndSig[0])}
+	}
+
+	txCount := 0
+	signer := types.MakeSigner(s.b.ChainConfig(), s.b.CurrentBlock().Number())
+	key, _ := crypto.GenerateKey()
+
+	signedBatches := make([][]*types.Transaction, len(blockBatches.Batches))
+	for bi, rollupTxs := range blockBatches.Batches {
+		signedBatches[bi] = make([]*types.Transaction, len(rollupTxs))
+		for i, rollupTx := range rollupTxs {
+			txCount++
+			tx := rollupTx.toTransaction()
+			tx, err := types.SignTx(tx, signer, key)
+			if err != nil {
+				return []error{fmt.Errorf("Error signing transaction in batch %d, index %d", bi, i)}
+			}
+			signedBatches[bi][i] = tx
+		}
+	}
+
+	s.b.SetTimestamp(int64(*blockBatches.Timestamp))
```

Comment:
We just chatted offline, but just want to ensure that all batches in `blockBatches` will be L1toL2TxBatches and SafetyTxBatches. Those batches are submitted with timestamp == L1 timestamp of the block it's submitted in. However w/ the sequencer batches the timestamp is !== L1 timestamp of the block its submitted in.

Reply:
Note another batch that can be submitted is txs inside of the `CanonicalTransactionChain` -- those also have their own timestamps associated with them

Reply:
I'm not sure this is a concern of Geth. If it's possible for this timestamp to not be accurate, it seems like it should be fixed where it's set in the Sync Service. Since we will definitely need _a_ timestamp, we should be fine to always set it here, right (even if it ends up being set to the same timestamp that we currently have)?

Reply:
Talked with @karlfloersch  about this, and it seems like I was misunderstanding what you were saying, @K-Ho. Looks like we may need to change the datatype to allow it to contain a timestamp per tx batch (multiple per `BlockBatches` type). I think the majority of those changes will still be upstream (in the Sync Service), but this is an important thing to account for -- good find!

Since I had to talk through a bunch of stuff on a few different occasions with Karl and Ben to be able to work on this workflow properly and Kevin had to talk through design with me (and Karl) to review the PR, this seems complex enough to justify documenting the overall flow. I think it'll be important to highlight:
* RollupTransaction Origin (likely party and queue it is submitted to / how is the associated event is triggered)
* Possible differences in content (quantity of RollupTransactions, timestamp etc.)
* How RollupTransactions from different sources may be handled together and how they must be handled separately
* Info that indicates source on the RollupTransaction itself to handle RollupTransactions from different sources differently downstream if necessary.